slash_command_register/register.py

```diff
@@ -32,17 +32,20 @@ def _build_dict_for_command(command: _command.Command) -> dict[str, typing.Any]:
 def _build_dict_for_option(
     command_option: _command.CommandOption,
 ) -> dict[str, typing.Any]:
-    return {
+    option_dict = {
         "name": command_option.name,
         "description": command_option.description,
         "type": command_option.type.value,
         "required": command_option.required,
-        "choices": [
-            choice.as_dict()
-            for choice in command_option.choices
-            if choice.type is command_option.type
-        ],
     }
+    if not command_option.choices:
+        return option_dict
+
+    if choices := [
+        choice.as_dict() for choice in command_option.choices if choice.type is command_option.type
```

Comment:
Do we need the `if choice.type is command_option.type` bit? Can we just assume that all choices are the correct type?

Reply:
I'd say yes because there's always a chance that someone could attempt to make, for example, a bool choice type which isn't valid. Even though its valid for the command, it's not for the choice type. So, if the choice is invalid, it's probably best that we allow the command to still work regardless. It could also help in future to keep the type in the choice if we need to parse a response back in future (at least in C# this would help, whether that's the case in Python idk but I think generally it's safer this way)

Reply:
>for example, a bool choice type which isn't valid. Even though its valid for the command, it's not for the choice type.

It feels like that's a programmer error though, and the code shouldn't have to deal with that -- we could add a test that takes all registered commands and checks their type is correct so no dev could merge a change that contains that kind of mistake

Reply:
Where would that go - `test_command_registry`? would it need to be it's own separate integration test? or a linter test?

Reply:
`test_command_registry` makes sense I think

Reply:
Feel free to merge this btw, we can change this to a test later, no need to hold up this PR

Reply:
ok